plugins/grpc-plugin/src/main.rs

```diff
@@ -34,18 +34,22 @@ async fn main() -> Result<()> {
             options::Value::Integer(-1),
             "Which port should the grpc plugin listen for incoming connections?",
         ))
-        .start()
+        .configure()
         .await?;
 
     let bind_port = match plugin.option("grpc-port") {
         Some(options::Value::Integer(-1)) => {
             log::info!("`grpc-port` option is not configured, exiting.");
+            plugin.disable("`grpc-port` option is not configured.").await?;
```

Comment:
So, if I'm understanding the API, this `disable` method gives you the possibility to change the method init.
Does this mean if in the future core lightning will need another parameter in the response, we need another method?

or a chain of methods because we need to join method payloads in only one result, right?

Reply:
For now we don't have other things to do, and I don't foresee much else being added to `init`, but your point is valid. This is much more about allowing user code to run after getting the `init` call, but before replying. It is equivalent to a callback, but avoids any limitation regarding context for a callback, and having additional state in the `ConfiguredPlugin` (i.e., the `disabled` field in your proposal), while still giving the ability to disable through a call. Notice that I made the `disable` method consume the `ConfiguredPlugin` making sure a user cannot accidentally call anything else after terminating this way.